SpatialGDK/Source/Private/Utils/ComponentReader.cpp

```diff
@@ -95,7 +95,39 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
 			{
 				if (Cmd.Type == REPCMD_DynamicArray)
 				{
-					ApplyArray(ComponentObject, FieldId, Cast<UArrayProperty>(Cmd.Property), Data, SwappedCmd.Offset, Cmd.ParentIndex);
+					UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Cmd.Property);
+					bool bProcessedArray = false;
+
+					if (UStructProperty* ParentStruct = Cast<UStructProperty>(Parent.Property))
```

Comment:
Could you add a comment with some context on what's going on in the following block?